EIPS/eip-7378.md

```diff
@@ -0,0 +1,75 @@
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-07-22
+---
+
+## Abstract
+
+The current formula for setting the base fee follows the original [EIP-1559](./eip-1559.md):
+$$b[i+1]\triangleq  b[i] \cdot \left( 1+\frac{1}{8} \cdot \frac{s[i]-s^* }{s^* }\right)$$ where $i$ enumerates the blocks, $s^*$ is a predetremined block size (the "desired" size, aka gas amount), $b[i]$ is the base fee at block $i$, and $s[i]$ is the size of the block at height $i$. This formula considers only the last block size $s[i]$. This mechanism might lead to incentives for users to bribe miners in order to reduce the base fee, or (in analogous manner) for miners to intiate a collusion with sophisticated users for the benefit of both. (See Motivation section.)
+
+We propose to consider the history of block sizes via a past wheighted average with a geometric sequence as the weights. In particular, we suggest the following update rule:
+$$b[i+1]\triangleq  b[i] \cdot \left( 1+\frac{1}{8} \cdot \frac{s_{\textit{avg}}[i]-s^* }{s^* }\right)$$ where $s_{\textit{avg}}[i]$ is defined by
+$$s_{\textit{avg}}[i] \triangleq \frac{1-q}{q}\sum_{k=1}^{\infty} q^k\cdot s[i-k+1], \hspace{1cm}   q\in (0,1).$$ 
+
+## Motivation
+
+Mainly, to reduce bribe motivation (in particular in case the demand for blockspace is high (see Incentive Considerations section). Additionaly, to reduce oscillations and have a more stable fee setting mechanism.
+
+For determining which messages enter a block, a miner uses a mechnism that was proposed in EIP-1559. This mechanism introduced a base fee, which is a portion of the transaction fee that is burned and not awarded to miners, and which varies according to the fill rate of blocks. A target block size is defined, such that if a block is fuller than the target size, the base fee increases and if it is emptier, the base fee reduces.
+
+Research on the subject have revealed issues with this transaction fee mechanism. It has been shown to be [unstable in cases](../assets/eip-7378/LMRSP.pdf). Moreover, it has been shown that the dynamic nature of the base fee, which is influenced by the fill rate of blocks, opens the door for [manipulation by miners and users](../assets/eip-7378/AGHH.pdf). The desired behavior of the system under a stable high demand, is for it to reach an equalibrium where the base fee -- $b$ -- is the significant part of the gas fee, and the tip is relatively small -- denoted $\varepsilon$ (for reference, Ethereum's base fee often has $\frac{b}{\varepsilon}\approx 20$). According to [Roughgarden](../assets/eip-7378/TR1559.pdf) this is a rational equalibrium under the assumption that miners do not think ahead. However, we expect a miner to optimize its behavior by also considering its future payoffs. In essence, since neither the miner nor the user are getting the burnt fee, by colluding they can both tap into the burnt fee for a win-win situation for them both.
```

Comment:
Actually, the analysis was even more conservative. To show the issues are in the core mechanism and not an implementation thing, the theoretical work modeled the randomness as perfect (e.g. unpredictable) and only considered expected gains vs. expected losses. In practice, due to the predictability (albeit limited), there should be fewer losses because an attacker might know in advance whether it will lose or profit.